ting-bill-split.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+)
+
+func isBadParam(p *string) bool {
+	return *p == ""
+}
+
+func checkParam() func(string, *string, *bool) {
```

Comment:
Since we aren't defining anything inside `checkParam` for the closure to close around this is actually creating unneeded complexity. Using a `*bool` for the result is a fine approach so here's what I'd say:

1. change `checkParam` to have the signature of the func it was returning 
2. Remove the `else` and just return early if the param is bad
3. Set the `badParam` inside the if block
4. Remove the `isBadParam` function and do the comparison inline. We don't need an extra func for that